spec/compiler/DocumentParserSpec.js

```diff
@@ -138,6 +143,19 @@ describe("DocumentParser", function() {
     expect(doc.title).toEqual('hello');
   });
 
+  it("adds metadata to document befor comment", function() {
+    var doc = parse(DOCUMENT_METADATA_1, COMMENT, HEAD, TITLE, HEAD_END);
+    expect(doc.metadata.experiment).toEqual(true);
+    expect(doc.sections.length).toEqual(1);
+  });
+
+  it("adds metadata to document after comment", function() {
+    var doc = parse(COMMENT, DOCUMENT_METADATA_2, HEAD, TITLE, HEAD_END, BODY, COMMENT, BODY_END);
+    expect(doc.metadata.experiment).toEqual(true);
+    expect(doc.metadata.component).toEqual("amp-accordion");
+    expect(doc.sections.length).toEqual(4);
+  });
+
```

Comment:
What happens if the JSON is invalid? Can we add a test for this case? I think it should fail loudly...